File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/audio/BluetoothAudioType.java
package com.mapbox.services.android.navigation.v5.navigation.metrics.audio;

import android.content.Context;
import android.media.AudioManager;

public class BluetoothAudioType implements AudioTypeResolver {
  private static final String BLUETOOTH = "bluetooth";
  private AudioTypeResolver chain;

  @Override
  public void nextChain(AudioTypeResolver chain) {
    this.chain = chain;
  }

  @Override
  public String obtainAudioType(Context context) {
    AudioManager audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
    if (audioManager == null) {
      return "unknown";
    }
    return audioManager.isBluetoothScoOn() ? BLUETOOTH : chain.obtainAudioType(context);
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/audio/BluetoothAudioType.kt
package com.mapbox.services.android.navigation.v5.navigation.metrics.audio

import android.content.Context
import android.media.AudioManager

internal class BluetoothAudioType : AudioTypeResolver {
    private val BLUETOOTH = "bluetooth"
    private var chain: AudioTypeResolver? = null

    override fun nextChain(chain: AudioTypeResolver) {
        this.chain = chain
    }

    override fun obtainAudioType(context: Context): String {
        val audioManager: AudioManager? = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
        return audioManager?.let { audio ->
            when (audio.isBluetoothScoOn) {
                true -> BLUETOOTH
                else -> {
                    chain?.obtainAudioType(context) ?: "unknown"
                }
            }
        }?:"unknown"
    }
}
